packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -0,0 +1,479 @@
+        }
+
+        if (!this.props.multiselectable && initialSelection.length > 1) {
+            initialSelection = initialSelection.slice(0, 1);
+        }
+
+        this.updateSelection(initialSelection);
+        this.toggleMenu(this.checkPropsForMenuState());
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                {...this.unhandledProps()}
+                aria-disabled={this.props.disabled || null}
+                ref={this.rootElement}
+                className={this.generateClassNames()}
+                onKeyDown={this.handleKeydown}
+                onClick={this.selectClicked}
+            >
+                {this.renderContentDisplay()}
+                {this.renderHiddenSelectElement()}
+                {this.renderMenu()}
+            </div>
+        );
+    }
+
+    /**
+     * Create class names
+     */
+    protected generateClassNames(): string {
+        let className: string = get(this.props.managedClasses, "select", "");
+
+        if (this.props.disabled) {
+            className = className.concat(
+                " ",
+                get(this.props.managedClasses, "select__disabled", "")
+            );
+        }
+
+        if (this.state.isMenuOpen) {
+            className = className.concat(
+                " ",
+                get(this.props.managedClasses, "select__menuOpen", "")
+            );
+        }
+
+        return super.generateClassNames(className);
+    }
+
+    /**
+     * Determine menu state based on props
+     */
+    private checkPropsForMenuState = (): boolean => {
+        let shouldMenuBeOpen: boolean = false;
+
+        if (this.props.isMenuOpen !== undefined) {
+            shouldMenuBeOpen = this.props.isMenuOpen;
+        } else if (this.props.multiselectable === true) {
+            shouldMenuBeOpen = true;
+        }
+
+        return shouldMenuBeOpen;
+    };
+
+    /**
+     * Renders a hidden select element which can interact with a
+     * form hosting this component
+     */
+    private renderHiddenSelectElement(): React.ReactNode {
+        return (
+            <select
+                required={this.props.required || null}
+                name={this.props.name || null}
+                form={this.props.form || null}
+                value={this.state.value}
+                multiple={this.props.multiselectable || null}
+                disabled={this.props.disabled || null}
+                onChange={this.onSelectValueChange}
+                style={{
+                    display: "none",
+                }}
+            />
+        );
+    }
+
+    /**
+     * This exists only to suppress a build warning
+     */
+    private onSelectValueChange = (event: React.ChangeEvent): void => {
+        return null;
+    };
+
+    /**
+     * Deternmines which function to use to render content display (ie. the part of the control that shows when the menu isn't open)
```

Comment:
nit: Determines* - you might want to run this file through spellcheck, I see this word spelt wrong a few times

Reply:
corrected